questions/javascript-questions.md

```diff
@@ -1,66 +1,54 @@
 # JS Questions:
 
-* Explain event delegation
-* Explain how `this` works in JavaScript
-* Explain how prototypal inheritance works
-* What do you think of AMD vs CommonJS?
-* Explain why the following doesn't work as an IIFE: `function foo(){ }();`.
-  * What needs to be changed to properly make it an IIFE?
+* Explain event delegation.
+* Explain how `this` works in JavaScript.
+  * Can you give an example of how `this` has changed in ES6?
+* Explain how prototypal inheritance works.
 * What's the difference between a variable that is: `null`, `undefined` or undeclared?
   * How would you go about checking for any of these states?
 * What is a closure, and how/why would you use one?
+* What language constructions do you use for iterating over object properties and array items?
 * Can you describe the main difference between a `forEach` loop and a `.map()` loop and why you would pick one versus the other?
 * What's a typical use case for anonymous functions?
-* How do you organize your code? (module pattern, classical inheritance?)
 * What's the difference between host objects and native objects?
-* Difference between: `function Person(){}`, `var person = Person()`, and `var person = new Person()`?
-* What's the difference between `.call` and `.apply`?
+* Explain the difference between: `function Person(){}`, `var person = Person()`, and `var person = new Person()`?
+* Explain the differences on the usage of `foo` between `function foo() {}` and `var foo = function() {}`
+* Can you explain what `.call` and `.apply` do? What's the notable difference between the two?
 * Explain `Function.prototype.bind`.
 * What's the difference between feature detection, feature inference, and using the UA string?
-* Explain Ajax in as much detail as possible.
```

Comment:
Do people even talk about Ajax anymore? You could have a decade of experience and not really deal with "Ajax" in the way that this question implies.

Reply:
For me this question is fine

Reply:
Maybe rephrase it to talk about the Fetch API instead, or ask about the differences between Fetch and Ajax?